addons/purchase_stock/models/stock.py

```diff
@@ -8,10 +8,18 @@
 class StockPicking(models.Model):
     _inherit = 'stock.picking'
 
+    # TODO in master, remove this field and use purchase_ids instead
     purchase_id = fields.Many2one(
         'purchase.order', related='move_ids.purchase_line_id.order_id',
+        string="Purchase Order", readonly=True)
+    # deprecating the field "purchase_id"
+    purchase_ids = fields.Many2many(
+        'purchase.order', compute='_compute_purchase_ids', relation='purchase_order_stock_picking_rel', store=True,
         string="Purchase Orders", readonly=True)
 
+    def _compute_purchase_ids(self):
+        self.purchase_ids = self.move_ids.purchase_line_id.order_id
```

Comment:
Doesn't this compute needs dependencies?

Reply:
Yes I forgot to add a dependency on move_ids. Added in the last push